Rules/AvoidUninitializedVariable.cs

```diff
@@ -65,13 +65,16 @@ public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
 
                 HashSet<string> paramVariables = new HashSet<string>();
 
-                if (isDscResourceFile && targetResourcesFunctions.Contains(funcAst.Name, StringComparer.OrdinalIgnoreCase))
+                // don't raise the rules for variables in the param block.
+                if (funcAst.Body != null && funcAst.Body.ParamBlock != null && funcAst.Body.ParamBlock.Parameters != null)
```

Comment:
I think you forgot to put back the isDscResourceFile boolean variable in the check?